query/src/util.rs

```diff
@@ -33,6 +33,20 @@ pub fn visit_expression<V: ExpressionVisitor>(expr: &Expr, visitor: &mut V) {
             visit_expression(right, visitor);
         }
         Expr::Cast { expr, .. } => visit_expression(expr, visitor),
+        Expr::Case {
+            expr,
+            when_then_expr,
+            else_expr,
+        } => {
+            expr.as_ref().map(|expr| visit_expression(expr, visitor));
```

Comment:
This throws a lint on my machine as the mapped value isn't used. We could write it as:

```suggestion
            if let Some(e) = expr.as_ref() {
                visit_expression(e, visitor)
            }
```